lib/libzpool/Makefile.am

```diff
@@ -202,3 +203,6 @@ libzpool_la_LIBADD += $(ZLIB) -ldl
 libzpool_la_LDFLAGS = -pthread -version-info 2:0:0
 
 EXTRA_DIST = $(USER_C)
+
+vdev_raidz_math_powerpc_altivec.$(OBJEXT): CFLAGS += $(if $(findstring powerpc,$(target_cpu)),-maltivec,)
+vdev_raidz_math_powerpc_altivec.l$(OBJEXT): CFLAGS += $(if $(findstring powerpc,$(target_cpu)),-maltivec,)
```

Comment:
Actually, one issue.  This code isn't portable and generates the following warning when included in the `Makefile.am`.

```
lib/libzpool/Makefile.am:207: warning: if $(findstring powerpc,$(target_cpu: non-POSIX variable name
lib/libzpool/Makefile.am:207: (probably a GNU make extension)
lib/libzpool/Makefile.am:208: warning: if $(findstring powerpc,$(target_cpu: non-POSIX variable name
lib/libzpool/Makefile.am:208: (probably a GNU make extension)
```

Reply:
@behlendorf Thanks for the testing & the numbers:

As for the Makefile.am, yes, it's far from ideal (point 6 in my original comments). My Makefile-fu is failing me here, I don't know how to do that 'cleanly' ('that' being: add the -maltivec option when compiling the AltiVec file on powerpc, but not on other platforms). Any suggestion on how to improve that welcome!

Reply:
I've opened PR #9848 with some proposed cleanup to facilitate the needed Makefile additions.  With those changes applied we can make the following Makefile changes.  Assuming there aren't any major downsides to adding `-maltivec` to more then just the new `vdev_raidz_math_powerpc_altivec.c` file.  If so, we can restrict it.

lib/libzpool/Makefile.am
```
# Required by vdev_raidz_math_powerpc_altivec.o
if TARGET_CPU_POWERPC
AM_CFLAGS += -maltivec
endif
```

For the kernel side using `CONFIG_ALTIVEC` to enable this feels more correct.

module/zfs/Makefile.in
```
# Required by vdev_raidz_math_powerpc_altivec.o
cflags-$(CONFIG_ALTIVEC) += -maltivec

...

$(MODULE)-$(CONFIG_ALTIVEC) += vdev_raidz_math_powerpc_altivec.o
```